cmd/site-replication.go

```diff
@@ -5510,3 +6061,100 @@ func (c *SiteReplicationSys) getSiteMetrics(ctx context.Context) (madmin.SRMetri
 	sm.Uptime = UTCNow().Unix() - globalBootTime.Unix()
 	return sm, nil
 }
+
+// mergeWithCurrentLCConfig - merges the given ilm expiry configuration with existing for the current site and returns
+func mergeWithCurrentLCConfig(ctx context.Context, bucket string, expLCCfg *string, updatedAt time.Time) ([]byte, error) {
```

Comment:
Is merge logic really needed ? We already require the newly added site replication to be empty, so I guess it is enough to overwrite the ILM policy.

Let's also forbid adding a new ILM transition rule when the site replication is enabled

Reply:
This merge is need for the below scenario.

Say we have 1 transition + 2 expiry rules under stite1 and then we enable site replication to site2 with `--replicate-ilm-expiry` flag. Now only the two expiry rules from site1 would be replicated to site2.

Later say we update contents of the expiry rules in site2. As the latest update time is higher on site2, these rules need to be replicated back to site1 as part of healing. Now while writing these expiry rules from site2 to site1, we should not overwrite all rules of site1 but merge with exiting ones on site1.

Without this merge, all transition rules on site1were getting lost as part of heal action.
As part of merge first we find all local rules, if incoming expiry rules already there in local list we update their content else just add to the local list. Then finally this list is returned for the local site for write.

There is no adding of any new ILM transition rule when site replication enabled.